Project 2/k-NNAlgorithms/src/Data.java

```diff
@@ -41,9 +41,9 @@ public void normalizeData(){
             }
         }
         //go through the data set and normalize values between 0-1
+        int indiceOfLastTrait = fullSet.get(0).size()-1;
         for (int i = 0; i <fullSet.size() ; i++) {
-
-            for (int j = 0; j <fullSet.get(0).size()-1 ; j++) {
+            for (int j = 0; j <indiceOfLastTrait; j++) {
```

Comment:
I know this is nitpicky now, but I'm pretty sure indice isn't a word, we should probably start using index.

Reply:
You know what you are right, I'm a dummy i'll fix it